src/ripplegw/interfaces.py

```
@@ -0,0 +1,374 @@
+from .waveforms.IMRPhenomD import gen_IMRPhenomD_hphc
+from .waveforms.IMRPhenomPv2 import gen_IMRPhenomPv2_hphc
+from .waveforms.TaylorF2 import gen_TaylorF2_hphc
+from .waveforms.IMRPhenomD_NRTidalv2 import gen_IMRPhenomD_NRTidalv2_hphc
+from .waveforms.IMRPhenomXAS import gen_IMRPhenomXAS_hphc
+from .waveforms.IMRPhenomXAS_NRTidalv3 import gen_IMRPhenomXAS_NRTidalv3_hphc
+from .waveforms.SineGaussian import gen_SineGaussian_hphc
+from .waveforms.IMRPhenomXPHM import generate_xphm
+from .conversions import Mc_eta_to_ms
+
+
+class Waveform(ABC):
```

Ruff (0.15.9)

[warning] `Waveform` is an abstract base class, but it has no abstract methods or properties

(B024)